### PR TITLE
fix(ui-react-native): Improve required fields handling for React Native Authenticator.

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

This change prevents undefined values from reaching the state machine, which causes internal errors to be exposed to end users.
Also adds a unit test to prevent re-occurrence.

| Before  | After |
| ------------- | ------------- |
| Simulator Screen Shot - iPhone 13 - 2022-12-19 at 10 44 03 | Simulator Screen Shot - iPhone 13 - 2023-01-17 at 12 55 14 |

#### Issue #, if available

#3190 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by joebuono — DISMISSED** [review]
LGTM

File: packages/react-native/src/Authenticator/hooks/useFieldValues/useFieldValues.ts [hunk 1]
```diff
@@ -70,13 +70,23 @@ export default function useFieldValues<FieldType extends TypedField>({
       setValues({ ...values, [name]: value });
     };
 
+    const getDisplayLabel = (): string | undefined => {
+      if (!label || labelHidden) return undefined;
+
+      // if * is already appended through state machine manipulations
+      if (label.endsWith('*')) return label;
```
Comment (ioanabrooks) [inline packages/react-native/src/Authenticator/hooks/useFieldValues/useFieldValues.ts:77]: Such as [here](https://github.com/aws-amplify/amplify-ui/blob/e25bc42693cc4fa1cdcf7ad2fe7034ff44fbb18e/packages/ui/src/helpers/authenticator/formFields/defaults.ts#L51)

**Review by calebpollman — APPROVED** [review]
🚢 (is this meant to be a draft 😄)

**Comment by ioanabrooks** [comment]
> 🚢 (is this meant to be a draft 😄)

Ah yes, sorry about it, temporarily converted to draft as I was re-scoping/slimming down the PR.

**Review by joebuono — APPROVED** [review]

**Pull request merged**